presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp

```diff
@@ -207,28 +207,27 @@ std::string PrestoServerOperations::taskOperation(
     }
     case ServerOperation::Action::kListAll: {
       uint32_t limit;
+      const auto& limitStr = message->getQueryParam("limit");
       try {
-        const auto& limitStr = message->getQueryParam("limit");
         limit = limitStr == proxygen::empty_string
             ? std::numeric_limits<uint32_t>::max()
             : stoi(limitStr);
       } catch (std::exception& ex) {
-        VELOX_USER_FAIL(ex.what());
+        VELOX_USER_FAIL("Invalid limit provided '{}'.", limitStr);
       }
       std::stringstream oss;
       oss << "[";
       uint32_t count = 0;
-      for (const auto& task : taskMap) {
-        const auto& veloxTask = task.second->task;
+      for (auto taskItr = taskMap.begin(); taskItr != taskMap.end();
+           ++taskItr) {
+        const auto& veloxTask = taskItr->second->task;
         if (++count > limit) {
-          oss << "... " << (taskMap.size() - limit) << " more tasks ...\n";
+          oss << "... " << (taskMap.size() - limit) << " more tasks ...]\n";
```

Comment:
Curious, how do you verify that this returns valid JSON. Would you share an example response? It feels like it is not a valid JSON.

Reply:
When limit is put, the option is to return an incomplete JSON with the ... n more tasks ... at the end to indicate the whole picture.

Reply:
But we can probably change it to return a JSON with limited tasks in it and a trailing statement saying n more tasks